api/src/config.ts

```diff
@@ -1,9 +1,29 @@
-const getEnvString = (name: string): string => process.env[name] || '';
-const getEnvNumber = (name: string): number => parseInt(getEnvString(name), 10);
-const getEnvBoolean = (name: string): boolean => getEnvString(name) === 'true';
+const getEnvString = (name: string, defaultValue?: string): string => {
+    let value = '';
+    try {
+        if (!(process.env[name]) && (defaultValue === undefined || defaultValue === '')) {
+            throw new Error('Environment variable is undefined');
+        } else if (!(process.env[name])) {
+            value = defaultValue as string;
```

Comment:
Why is `as string` needed here?

Reply:
Typescript will complain because it does not know the type of the `defaultValue` variable since it is an optional variable. It will also complain about the type of the `process.env[name]` variable. I think of it as _casting_ the variable to type `string`.